sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1906,353 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+class QuantileNormalizer(BaseEstimator, TransformerMixin):
+    """Normalize features using quantiles information.
+
+    This Normalizer scales the features between 0 and 1, equalizing the
+    distribution of each feature to a uniform distribution. Therefore,
+    for a given feature, this normalization tends to spread out the most
+    frequent values.
```

Comment:
It also reduces the impact of (marginal) outliers: this is therefore a robust preprocessing scheme.